lib/kitchen/driver/azurerm.rb

```diff
@@ -376,9 +374,10 @@ def existing_state_value?(state, property)
       # @return [Hash] Updated Hash of state values.
       def validate_state(state = {})
         state[:uuid] = SecureRandom.hex(8) unless existing_state_value?(state, :uuid)
+        state[:vm_name] = config[:vm_name] || "tk-#{state[:uuid][0..11]}" unless existing_state_value?(state, :vm_name)
```

Comment:
I figured I should just use the state uuid, but not sure if that's the "best" thing to do here or if this should be a separately generated string. Please let me know your thoughts.